pkgs/servers/sql/cockroachdb/generic.nix

```diff
@@ -0,0 +1,63 @@
+{ lib, stdenv, buildGoModule, fetchurl, cmake, xz, which, autoconf, ncurses6
+, libedit, libunwind, installShellFiles, removeReferencesTo, go, version, sha256
+}:
+
+buildGoModule rec {
+  pname = "cockroach";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://binaries.cockroachdb.com/cockroach-v${version}.src.tgz";
+    inherit sha256;
+  };
+  vendorSha256 = null;
+
+  NIX_CFLAGS_COMPILE = lib.optionals stdenv.cc.isGNU [
+    "-Wno-error=deprecated-copy"
+    "-Wno-error=redundant-move"
+    "-Wno-error=pessimizing-move"
+  ];
+
+  nativeBuildInputs = [ installShellFiles cmake xz which autoconf ];
+  buildInputs = if stdenv.isDarwin then [ libunwind libedit ] else [ ncurses6 ];
+
+  postPatch = ''
+    patchShebangs .
+  '';
+  buildPhase = ''
+    runHook preBuild
+    make buildoss
+    cd src/github.com/cockroachdb/cockroach
+    for asset in man autocomplete; do
+      ./cockroachoss gen $asset
+    done
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D cockroachoss $out/bin/cockroach
+    installShellCompletion cockroach.bash
+
+    installManPage man/man1/*
+
+    runHook postInstall
+  '';
+
+  outputs = [ "out" "man" ];
+
+  # fails with `GOFLAGS=-trimpath`
+  allowGoReference = true;
+  preFixup = ''
+    find $out -type f -exec ${removeReferencesTo}/bin/remove-references-to -t ${go} '{}' +
+  '';
+
+  meta = with lib; {
+    homepage = "https://www.cockroachlabs.com";
+    description = "A scalable, survivable, strongly-consistent SQL database";
+    license = licenses.bsl11;
+    platforms = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" ];
+    maintainers = with maintainers; [ rushmorem thoughtpolice turion ];
```

Comment:
If you're fine with it you can add yourself here :)

Reply:
I'm going to leave myself out for the time being. I'm doing this update in order to try cockroach out on a project. If I end up using it even semi-seriously, I will add myself to the list of maintainers.